### ISSUE TITLE
bug: Accessing field with NULL value in a VARIANT from PARSE_JSON returns the whole object instead.

### ISSUE THREAD

**Issue by dixel** [datafuselabs/databend#14782]
### Search before asking

- [X] I had searched in the [issues](https://github.com/datafuselabs/databend/issues) and found no similar issues.

### Version

DatabendQuery v1.2.279-638686a9ae(rust-1.75.0-nightly-2024-01-02T10:52:45.307518102Z)

### What's Wrong?

When accessing fields in a VARIANT that are NULL, the whole object instead of NULL is returned.

### How to Reproduce?

```sql
SELECT PARSE_JSON('{"x": null}'):x;

SELECT
  PARSE_JSON('{"x": null}') :x

┌─────────────────────────────┐
│ parse_json('{"x": null}'):x │
│      Nullable(Variant)      │
├─────────────────────────────┤
│ {"x":null}                  │
└─────────────────────────────┘
1 row read in 0.033 sec. Processed 1 row, 1 B (30.4 row/s, 30 B/s)
```

```sql
SELECT PARSE_JSON('{"x": null}')['x'];

SELECT
  PARSE_JSON('{"x": null}') ['x']

┌────────────────────────────────┐
│ parse_json('{"x": null}')['x'] │
│        Nullable(Variant)       │
├────────────────────────────────┤
│ {"x":null}                     │
└────────────────────────────────┘
1 row read in 0.050 sec. Processed 1 row, 1 B (20.04 row/s, 20 B/s)
```

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

**Comment by b41sh** [comment]
Hi, @dixel Thanks for reporting the bug, it has been fixed in PR #14684 , please download the latest version of databend to use.

**Comment by dixel** [comment]
Thank you @b41sh !
